docs/developers/developer-reference/sc-payments.md

```diff
@@ -0,0 +1,143 @@
+:::
+
+
+If an endpoint only accepts EGLD, it should be annotated with `#[payable("EGLD")]`:
+
+```rust
+#[endpoint]
+#[payable("EGLD")]
+fn accept_egld(&self) {
+	// ...
+}
+```
+
+When annotated like this, the contract will reject any ESDT payment. Calling this function without any payment will work.
+
+To accept any kind of payment, do annotate the endpoints with `#[payable("*")]`:
+
+```rust
+#[endpoint]
+#[payable("*")]
+fn accept_any_payment(&self) {
+	// ...
+}
+```
+
+:::note Hard-coded token identifier
+It is also possible to hard-code a token identifier in the `payable`, e.g. `#[payable("MYTOKEN-123456")]`. It is rarely, if ever, used, tokens should normally be configured in storage, or at runtime.
+:::
+
+Additional restrictions on the incoming tokens can be imposed in the body of the endpoint, by calling the call value API. Most of these functions retrieve data about the received payment, while also stopping execution if the payment is not of the expected type.
+- `self.call_value().egld_value()` retrieves the EGLD value transfered, or zero. Never stops execution.
+- `self.call_value().all_esdt_transfers()` retrieves all the ESDT transfers received, or an empty list. Never stops execution.
+- `self.call_value().multi_esdt<N>()` is ideal when we know exactly how many ESDT transfers we expect. It returns an array of `EsdtTokenPayment`. It knows exactly how many transfers to expect based on the return type (it is polymorphic in the length of the array). Will fail execution if the number of ESDT transfers does not match.
+- `self.call_value().single_esdt()` expects a single ESDT transfer, fails otherwise. Will return the received `EsdtTokenPayment`. It is a special case of `multi_esdt`, where `N` is 1.
+- `self.call_value().single_fungible_esdt()` further restricts `single_esdt` to only fungible tokens, so those with their nonce zero. Returns the token identifier and amount, as pair.
+- `self.call_value().egld_or_single_esdt()` retrieves an object of type `EgldOrEsdtTokenPayment`. Will halt execution in case of ESDT multi-transfer.
+- `self.call_value().egld_or_single_fungible_esdt()` further restricts `egld_or_single_esdt` to fungible ESDT tokens. It will return a pair of `EgldOrEsdtTokenIdentifier` and an amount.
+- `self.call_value().any_payment()` is the most general payment retriever. Never stops execution. Returns an object of type `EgldOrMultiEsdtPayment`.
+
+
+
+---
+
+[comment]: # (mx-context-auto)
+
+## Sending payments
+
+
+We have seen how contracts can accomodate receiving tokens. Sending them is, in principle, even more straightforward, as it only involves calling methods from the `self.send()` component.
+
+
+[comment]: # (mx-context-auto)
+
+### Sending payments directly
+
+Contracts can send tokens directly, without calling any endpoint at the destination, by using several methods from the API.
```

Comment:
If the contract is payable by default, otherwise it's disallowed.

Reply:
This is about sending, not receiving. We talked about direct transfers and payable contracts in a different section.